### DISCUSSION TITLE
Enable auto-suspend for PocketBook InkPad 3

### DISCUSSION THREAD

**Discussion by seb-maillard** [koreader/koreader#12791] [Q&A] [answered]
Hi there,

### Question
I wasn't happy with the stock InkPad _Auto screen lock_ settings (Off, 5 and 10 minutes) so I looked into enabling _Timeout for autosuspend_ from the _Auto power save_ KOReader plugin. Indeed, from what I understand, **suspend** [is equivalent](https://github.com/koreader/koreader/issues/10117#issuecomment-1429708968) to **screen lock** a.k.a. **sleep mode** on InkPad. However, this plugin is not enabled on my InkPad 3. Is there any reason why?

### Standby vs. suspend

I'm assuming this is why the _Auto Standby_ feature [was implemented](https://github.com/koreader/koreader/pull/6559) in the first place. However, leaving the device idle slowly drains the battery (e.g. 5~10% per 24 hours with frontlight off), despite the _Auto Standby_ plugin being enabled, whereas if I manually put the device to sleep, it virtually doesn't use any battery (less than 1% per 24 hours). Which is why I'd like to set my device to automatically **lock/go to sleep** after let's say 60 minutes of inactivity.

I'm actually a bit confused by some of the wording [here](https://github.com/koreader/koreader-base/blob/509f8090d74d699a54483cce47ca96ad98baea2b/ffi/input_pocketbook.lua#L425-L426):

> We have got sleepmode ("autostandby") requested by frontend. We must be a "properly" foregroun running to do this - no keylock (zzzz)

It refers to **sleep mode** as **standby**, when I assumed sleep mode is equivalent to **suspend**. Also, it mentions not triggering "zzzz", which is another way to call the sleep mode as it refers to the **screen lock** [overlay](https://www.mobileread.com/forums/showpost.php?p=4342734&postcount=1).

### What I've looked into so far
Considering the InkPad natively features a screen lock/sleep mode (i.e. can suspend), I tried enabling the `AutoSuspend` plugin (e.g. by disabling [this check](https://github.com/koreader/koreader/blob/798a64998fd8c785478fbba09df16cecf41b23f3/plugins/autosuspend.koplugin/main.lua#L4-L6) and/or by [setting](https://github.com/koreader/koreader/blob/798a64998fd8c785478fbba09df16cecf41b23f3/frontend/device/pocketbook/device.lua#L32) `canSuspend` to `yes`). I set a value for _Timeout for autosuspend_ in the menu, but unfortunately the device never automatically went to sleep, so I'm thinking the actual suspend procedure is maybe not implemented.

I should mention Wi-Fi was disabled and USB was unplugged while I did those tests, as [this seems to be](https://github.com/koreader/koreader/blob/798a64998fd8c785478fbba09df16cecf41b23f3/plugins/autosuspend.koplugin/main.lua#L266-L277) a prerequisite to enable standby/suspend.

### Edit (2024-11-24)
I was reading [this thread](https://github.com/koreader/koreader/issues/3561) and this inspired me to enable the power button LED in the settings to monitor the standby behavior. When the battery level is below the _Always standby if battery below_ threshold, a.k.a. _If battery is below this percent, make auto-standby aggressive again (disables scaling by `mul`)_, the LED indeed turns off pretty much instantly.

However, **if the battery is above that threshold, the LED never seems to turn off,** i.e. the device doesn't actually seem to go into standby mode. I'm using the default settings and I waited for over 15 minutes after starting KOReader and opening a book, so I should in any case be way past the "scaling" timer. I was assuming from the _Max input idle seconds_ setting a.k.a. _Input that happens further than 30 seconds since last input one reset delay back to '`min`'_ that the device would automatically standby after 31 seconds of inactivity. Is there something I don't understand regarding how the plugin works, or should I create an issue to report this problem?

My initial question was about getting my device to automatically enter sleep mode since standby seemed to drain the battery. But if the device was actually not standing by when I made the test, my question would be void. I've set the _Always standby if battery below_ threshold to 100% and I will let it rest idle for a while, this time actually standing by, to check if it still drains battery compared to the sleep mode.

**Comment by seb-maillard** [comment]
Tagging @ezdiy since you seem to be the person behind the InkPad automatic standby implementation 😇

Edit: Even more relevant now with the question related to standby, see _Edit (2024-11-24)_ section above!

Note: Since I struggled a bit to understand the settings for the Auto Standby plugin, I created [a documentation](https://github.com/koreader/koreader/wiki/Auto-Standby) based off what was available [in the code](https://github.com/koreader/koreader/blob/798a64998fd8c785478fbba09df16cecf41b23f3/plugins/autostandby.koplugin/main.lua#L33-L38).

**Comment by NiLuJe** [comment]
Standby handling on PB is due a rewrite to behave like on Kobo: *we* control when to standby, instead of being at the mercy of InkView's blackbox-y timers.

TL;DR: It's a mess, see the numerous open PB issues about standby-related crap.

Help very much wanted, as none of us have access to the platform (nor do we want to, to be quite frank).

**Comment by NiLuJe** [comment]
c.f., #9493 for more context.

**Comment by seb-maillard** [comment]
Thanks @NiLuJe for the prompt feedback! Indeed, the thread you pointed me to seems to partially answer my questions.

Considering the standby implementation on the PB is crap, the bug I mentioned (standby is not activated when the battery is over the threshold) probably won't be resolved until the rewrite you mentioned. I'll send a message in the thread to see if I can be of some help in that regard.

One point I didn't get is why I can't enable the `AutoSuspend` plugin. @rjd22 [mentioned](https://github.com/koreader/koreader/issues/9493#issuecomment-1328198502) he would submit a PR to remove this plugin, but it seems the submission is still pending.
To be clear, assuming `AutoSuspend` does indeed put the device into sleep mode based on the _Timeout for autosuspend_ setting, this would actually be precisely the feature I'm looking for :)

**Reply by NiLuJe** [reply, marked as answer]
AFAIU `AutoSuspend` is currently locked behind having a rooted device (and running on a device where raw input has been tested, of which there's a grand total of one, the one it was implemented on).

(i.e., the `canSuspend` Device cap).

**Reply by seb-maillard** [reply]
Ok, thanks for the feedback! I guess I'll try my luck with [helping](https://github.com/koreader/koreader/issues/9493#issuecomment-2496117020) fixing the standby mode then.

**Comment by seb-maillard** [comment]
> standby is not activated when the battery is over the threshold

I did a bit of digging regarding that issue by adding some debug logs in the `autostandby` plugin ~~and it seems that the [`AutoStandby:allow()`](https://github.com/koreader/koreader/blob/798a64998fd8c785478fbba09df16cecf41b23f3/plugins/autostandby.koplugin/main.lua#L126) function is never triggered by the scheduler, even though it has actually [been scheduled](https://github.com/koreader/koreader/blob/798a64998fd8c785478fbba09df16cecf41b23f3/plugins/autostandby.koplugin/main.lua#L114). The only case where it's triggered is when it's [directly invoked](https://github.com/koreader/koreader/blob/798a64998fd8c785478fbba09df16cecf41b23f3/plugins/autostandby.koplugin/main.lua#L88) in case the battery level is below the threshold. That matches the behavior I described above.~~

### Edit
I was disabling debug logs after the tests I did above and I realized the LED turned off multiple times after a brief period of inactivity while I was navigating the menus. I did some more tests and the bug appears to be a bit trickier than I initially described it.

Indeed, after a fresh restart of KOReader, the device goes into standby as expected but **only while I'm in still in the file browser** and navigating in the menus, i.e. [`AutoStandby:allow()`](https://github.com/koreader/koreader/blob/798a64998fd8c785478fbba09df16cecf41b23f3/plugins/autostandby.koplugin/main.lua#L126) function is triggered by the scheduler after the [scheduled](https://github.com/koreader/koreader/blob/798a64998fd8c785478fbba09df16cecf41b23f3/plugins/autostandby.koplugin/main.lua#L114) amount of time. However, **as soon as I open a first file**, the scheduler stop triggering `AutoStandby:allow()`, even though it has been scheduled. And after I go back to the file browser, the scheduler still won't trigger it ever again, until the next restart of KOReader.
In all of my previous tests, I was systematically opening a file as soon as I would start KOReader, which explains why I didn't notice this behavior. 

@pazos Since [you mentioned](https://github.com/koreader/koreader/issues/9493#issuecomment-1257222684) tweaking the timers, maybe you have any idea why the function is never invoked by the scheduler after opening a file?

Since the plugin is device independent, should I create a dedicated issue to look into this problem? Or do we assume that the plugin will be deprecated after the [PB standby rewrite](https://github.com/koreader/koreader/issues/9493#issuecomment-1257544085)?

For now, I will set the battery threshold to 100% to enable the [aggressive standby mode](https://github.com/koreader/koreader/blob/798a64998fd8c785478fbba09df16cecf41b23f3/plugins/autostandby.koplugin/main.lua#L87). And I will provide an update on the battery drain in standby mode I mentioned in my first message after I'm done with the 24 hours idle test.

**Reply by pazos** [reply]
@sebastien-maillard: You're talking about low level stuff. Plenty of them, involving various plugins.

Please explain us what you're trying to achieve in secular terms, without terminology from within this repo. That way we can help you better :)

**Reply by pazos** [reply]
FWIW pocketbook is a very weird platform, pretty much like android. Instead of having full root capabilities on a barebones linux system we have to use middleware to do relevant stuff, such as read input events and write to the framebuffer.

**Reply by seb-maillard** [reply]
Thanks for the feedback and the context @pazos! And sorry for referencing so many concepts that I'm indeed not familiar with. I tried my best at piecing this information together. Here's the most simple way I can put it:

1. Initially, I was hoping I could get my device to enter sleep mode a.k.a. screen lock a.k.a. "Zzzz" after a certain period of inactivity (say 60 minutes) since I noticed it was slowly draining the battery when staying idle (i.e. when I don't interact with the device for an extensive period of time). According to @NiLuJe's [answer](https://github.com/koreader/koreader/discussions/12791#discussioncomment-11364187), it appears it's not possible using the `AutoSuspend` plugin.
2. After running some tests, it appeared my device was draining the battery when it was idle because it wasn't actually entering the standby mode. I realized that by enabling the power button LED indicator. Basically, as soon as I open a first book, the LED will never turn off. It stays on forever until I restart KOReader. Before opening a first book, the LED turns off shortly after I stop interacting with the file browser and the menus. Which makes me think I ran into some kind of bug.

**Reply by seb-maillard** [reply]
> And I will provide an update on the battery drain in standby mode I mentioned in my first message after I'm done with the 24 hours idle test.

I ran a bunch of additional battery drain tests, here are the results in case someone might be interested (with frontlight off):
- Device in sleep mode a.k.a. screen lock a.k.a. "Zzzz": less than 1% per 24 hours
- Device idle in actual standby mode (thanks to [this patch](#discussioncomment-11383332), confirmed with LED turning off): around 6% per 24 hours
- Device idle, standby disabled (i.e. `AutoStandby` plugin enabled and _Allow auto-standby_ checked off, _LED Indication_ set to _off_ in stock InkPad settings in order not to influence the results): 100% of battery drain within around 20 hours

**Comment by NiLuJe** [comment]
> Initially, I was hoping I could get my device to enter sleep mode a.k.a. screen lock a.k.a. "Zzzz" after a certain period of inactivity

As far as I understand, the onus is on the stock system to handle that (i.e., suspend) on PB, and I don't *think* we do anything to prevent that. We definitely don't do anything to explicitly control the timer, at least. But, again, not familiar with the platform, there may be weird InkView crap involved that I'm forgetting.

> it wasn't actually entering the standby mode.

Again, the onus is on the stock system, except the AutoStandby plugin may be preventing it from kicking in.

Not being able to enter standby may in turn also block entering suspend. Not sure, unfamiliar with the platform.

You can check verbose debug logs to check what the hell is happening with the AutoStandby plugin, at least, but expect a bumpy and confusing ride.

**Reply by seb-maillard** [reply]
> As far as I understand, the onus is on the stock system to handle that (i.e., suspend) on PB, and I don't think we do anything to prevent that. We definitely don't do anything to explicitly control the timer, at least.

Indeed, as I mentioned in my first [message](#discussion-7541759), the PB OS can put the device to sleep after a period of inactivity, but the options are too limited (5 and 10 minutes). Rather than controlling that "native" timer, I was hoping KOReader could send the device into sleep mode by invoking some kind of native function after detecting a period of inactivity, similarly to what the `AutoSuspend` plugin does e.g. for suspend and shutdown, but it seems this might not be possible.
I guess I'll just keep it disabled and rely on Auto standby to save battery in case I forget to manually put the device to sleep.

> Again, the onus is on the stock system, except the AutoStandby plugin may be preventing it from kicking in.
Not being able to enter standby may in turn also block entering suspend. Not sure, unfamiliar with the platform.

As far as my tests go, the OS is actually able to send the device to sleep even after KOReader has put it in standby, assuming the OS is configured to do so.

> You can check verbose debug logs to check what the hell is happening with the AutoStandby plugin, at least, but expect a bumpy and confusing ride.

Thanks for the cheer up! I'll try to look into that to understand why opening a book suddenly and completely stops the plugin from sending the device into standby. As I mentioned [above](#discussioncomment-11364565), I did add some extra logs in the plugin and I can confirm it does in fact [correctly schedule](https://github.com/koreader/koreader/blob/798a64998fd8c785478fbba09df16cecf41b23f3/plugins/autostandby.koplugin/main.lua#L114) the standby for when the delay expires. So I'm guessing something must be wrong with the `UIManager`'s ability to execute scheduled tasks in some cases.

Overall, I have to say that the 100% battery threshold trick is actually a great workaround. The device goes into standby almost instantly even when a book is open (that's because [`allow()`](https://github.com/koreader/koreader/blob/798a64998fd8c785478fbba09df16cecf41b23f3/plugins/autostandby.koplugin/main.lua#L88) is triggered directly instead of being scheduled), which is exactly how the OS handles standby in the stock applications. To be honest, the whole delay/schedule part of the plugin feels a bit overkill since the device is actually able to wake up instantly from standby.

**Reply by pazos** [reply]
> I was hoping KOReader could send the device into sleep mode by invoking some kind of native function after detecting a period of inactivity, similarly to what the AutoSuspend plugin does e.g. for suspend and shutdown, but it seems this might not be possible.

That's not possible. At least not directly. The Application Processor is dead when it enters standby. It cannot execute any codepath. So it isn't possible to schedule things for the future using application timers.

> To be honest, the whole delay/schedule part of the plugin feels a bit overkill since the device is actually able to wake up instantly from standby.

It isn't overkill. It was very well designed to provide a workaround for things in the future and provides mostly sane defaults. Your suggestion, on the other hand, would make https://github.com/koreader/koreader/issues/9493 much worse. It shouldn't be an issue for you if you don't rely on timers or you don't care about notifications stuck or ommited and any other kind of crap that happens when the application has scheduled task it isn't able to execute.

**Comment by seb-maillard** [comment]
> That's not possible. At least not directly. The Application Processor is dead when it enters standby. It cannot execute any codepath. So it isn't possible to schedule things for the future using application timers.

I see! Though I have to add that it seems the device is automatically waking up from standby precisely every 5 minutes. At least, the LED very briefly turns on and off. Since I haven't seen anything in the code of the plugin that would explain that, I'm assuming this is maybe some kind of native behavior, probably to check if the device needs to be put to sleep or something similar.

But indeed, it appears KOReader is not able to trigger the pending scheduled events when that happens. For instance, I tested the `AutoDim` plugin and set it to dim the light after 3 minutes of inactivity. Once standby, not only it won't dim the light after 3 minutes of inactivity, but it also won't dim it when automatically waking up from standby every 5 minutes.

However, every e.g. half-hour (but this can vary greatly), the status bar will refresh (e.g. time and battery level will update). I'm guessing some scheduled events actually do get triggered. But since the device wakes up very briefly (definitely less than a second), this is most likely not enough for most pending events to get triggered.

TL;DR: it does seem I ran into the standby shenanigans @NiLuJe mentioned [above](#discussioncomment-11363725) ;)

> It isn't overkill. It was very well designed to provide a workaround for things in the future and provides mostly sane defaults. Your suggestion, on the other hand, would make https://github.com/koreader/koreader/issues/9493 much worse. It shouldn't be an issue for you if you don't rely on timers or you don't care about notifications stuck or ommited and any other kind of crap that happens when the application has scheduled task it isn't able to execute.

Very good point indeed, I definitely didn't have such use-cases in mind. I actually used that to implement the `AutoDim` use-case from above, e.g.:

- _Min input idle seconds_ = 200, to give enough time for the dim process to run before going into standby
- _Input window seconds_ = 0, to send the device to standby right after the timer is reached
- _Max input idle seconds_ = 200, since it should be >= `min` + `win`

Obviously, that's assuming the device actually goes into standby eventually, which is not the case for me after I've opened a first book from the file browser.

**Comment by seb-maillard** [comment]
> After running some tests, it appeared my device was draining the battery when it was idle because it wasn't actually entering the standby mode. I realized that by enabling the power button LED indicator. Basically, as soon as I open a first book, the LED will never turn off. It stays on forever until I restart KOReader. Before opening a first book, the LED turns off shortly after I stop interacting with the file browser and the menus. Which makes me think I ran into some kind of bug.

So I was able to narrow down the bug a bit further:
- The device will never go into standby if I open a book and stay on the current page forever. As soon as I perform an action, such as turn a page, open the menu, touch a hot-corner, etc., the device will go into standby after the expected amount of time.
- And once I go back to the file browser, the same thing happens. The device will never go into standby until I perform an action, such as change folder, go to next page, open the menu, etc.

I only noticed it now because all my files are one-pager PDFs (I use my reader for music sheets). Basically, my workflow is always:
1. open a PDF file
2. leave it open on the same page for a while
3. go back to the file browser
4. open another file
5. and so on

It appears to be the perfect (sic) workflow to trigger this bug and ensure my device never goes into standby 🤦‍♂️

EDIT:
"Fun" fact, I noticed that when I go back to the file browser using the physical _Home_ button or by touching the screen, i.e. reach the end of the book (I've set _End of document action_ to _Return to file browser_ ), then the device will never go into standby. However, if I press the _Next page_ physical button, which should be the same as touching the screen (i.e. reach the end of the book), then the device goes into standby in the file browser after the expected amount of time.

**Reply by Frenzie** [reply]
The file browser from KOReader or PocketBook?

**Reply by seb-maillard** [reply]
KOReader's

**Reply by seb-maillard** [reply]
> You can check verbose debug logs to check what the hell is happening with the AutoStandby plugin, at least, but expect a bumpy and confusing ride.

Great news, I think I'm onto something!

I noticed in the logs that every time I open and close a book, [`AutoStandby:onCloseWidget()`](https://github.com/koreader/koreader/blob/f5c6b5689942b3e6c3ced1b33dcfc98ba482df22/plugins/autostandby.koplugin/main.lua#L47) is triggered. This happens right after `AutoStandby:allow()` [is scheduled](https://github.com/koreader/koreader/blob/f5c6b5689942b3e6c3ced1b33dcfc98ba482df22/plugins/autostandby.koplugin/main.lua#L114C5-L114C76). So it appears that `AutoStandby:allow()` is simply unscheduled by `AutoStandby:onCloseWidget()`.

Simplified chain of events when opening a book:

1. Press down on a book in the file browser
```
DEBUG PB event => 30 (EVT_POINTERDOWN)
DEBUG AutoStandby: schedule standby re-enable once delay expires
```
2. Release the finger
```
DEBUG PB event => 29 (EVT_POINTERUP)`
AutoStandby: input packed too close to previous one, ignoring
```
3. Book opens
```
DEBUG show reader ui
DEBUG close filemanager
DEBUG AutoStandby:onCloseWidget() instance= table: 0xa027abb0
INFO  opening file
DEBUG AutoStandby:init() instance= table: 0x9e7a9e78
DEBUG show widget: ReaderUI
```
4. Standby in not triggered, i.e. no corresponding `UIManager:allowStandby, counter decreased to 0` entry in the logs

The chain of events is pretty much the same when closing the book, i.e. `AutoStandby:onCloseWidget()` is triggered while returning to the file browser.

In contrast, here's a chain of event when interacting with a book or with the file browser:

1. Press down a hot-corner 
```
DEBUG PB event => 30 (EVT_POINTERDOWN)
DEBUG AutoStandby: schedule standby re-enable once delay expires
```
2. Release the finger
```
DEBUG PB event => 29 (EVT_POINTERUP)
AutoStandby: input packed too close to previous one, ignoring
```
3. Wait the expected delay
```
DEBUG UIManager:allowStandby, counter decreased to 0
DEBUG UIManager:_standbyTransition -> AllowStandby
DEBUG AutoStandby: onAllowStandby()
```

In light of the above, I deactivated the unscheduling of `AutoStandby:allow()` in `AutoStandby:onCloseWidget()` by commenting out the [corresponding line](https://github.com/koreader/koreader/blob/f5c6b5689942b3e6c3ced1b33dcfc98ba482df22/plugins/autostandby.koplugin/main.lua#L49). And it seems like the bug is now fixed 🎉

I tried a bunch of use-cases and it seems standby is working as expected now. To be clear, this doesn't solve any of the #9493 shenanigans! This is only a fix for the bug I reported earlier (device doesn't go into standby after opening or closing a book).
Since I don't really have any experience with neither KOReader, its code-base, the `AutoStandby` plugin and PB devices, I can't really tell if not unscheduling `AutoStandby:allow()` when closing the widget could have some undesirable side-effects. So I'd rather rely on some expert opinions before submitting a PR!

**Comment by NiLuJe** [comment]
IIRC, it's on purpose to prevent standby during a view transition. This
should probably be tackled the other way around, i.e., make sure standby is
allowed on init, even before the first input event. Might still be a
terrible idea, though (e.g., progress sync on book open?)

**Reply by seb-maillard** [reply]
> IIRC, it's on purpose to prevent standby during a view transition. This should probably be tackled the other way around, i.e., make sure standby is allowed on init, even before the first input event.

That makes sense! If the plugin is configured with a very aggressive setting (e.g. `min`, `win` and `max` = `0`), standby could indeed be scheduled during the transition.

> Might still be a terrible idea, though (e.g., progress sync on book open?)

I'm not sure what you're referring to, so I'll let others weigh in here.

One point in favor of scheduling the standby on init though: as soon as KOReader starts, the device goes instantly into standby. I'm guessing this is because the plugin doesn't prevent from going into standby when it's being initialized. So triggering part of the code of `AutoStandby:onInputEvent()` on init would both prevent the device from instantly going into standby and schedule a standby for when it's actually due.

Since calling `AutoStandby:onInputEvent()` on init would actually likely result in [filtering out](https://github.com/koreader/koreader/blob/f5c6b5689942b3e6c3ced1b33dcfc98ba482df22/plugins/autostandby.koplugin/main.lua#L76-L80) the call (because the transition happens right after pressing the screen), I isolated [this section](https://github.com/koreader/koreader/blob/f5c6b5689942b3e6c3ced1b33dcfc98ba482df22/plugins/autostandby.koplugin/main.lua#L82-L114) in a separate function which I called in [`AutoStandby:init()`](https://github.com/koreader/koreader/blob/f5c6b5689942b3e6c3ced1b33dcfc98ba482df22/plugins/autostandby.koplugin/main.lua#L27C10-L27C28). This does seem to work, i.e. the bug I reported is fixed, the device doesn't instantly go into standby when KOReaders starts and standby behaves as expected the rest of the time.

**Reply by seb-maillard** [reply]
Despite all the other issues caused by the `AutoStandby` plugin (see #9493), considering that allowing the device to go into standby actually has [a huge impact](#discussioncomment-11406502) on battery life, let me know if I should submit a patch to fix the issues mentioned in my previous comment. Due to my lack of experience with Lua and despite being a rather simple fix, beware it might need a bit of rework before it can be merged!

**Reply by pazos** [reply]
>  let me know if I should submit a patch to fix the issues mentioned in my previous comment

Autostandby improvements are welcome. Also documentation is very welcome.

> Due to my lack of experience with Lua and despite being a rather simple fix, beware it might need a bit of rework before it can be merged!

That's not a problem here :). Go ahead and submit a PR with the minimal changes required. Make sure you test your changes against platform defaults (ie: without changing any parameter within the plugin)

**Reply by seb-maillard** [reply]
> Autostandby improvements are welcome

Awesome!
- I created the issue #12814 to summarize the bugs I reported during this discussion
- I submitted the PR #12815  in an attempt to fix those bugs

> That's not a problem here :). Go ahead and submit a PR with the minimal changes required.

Thanks! I added a _Disclaimer_ section in the PR to highlight those points.

> Make sure you test your changes against platform defaults (ie: without changing any parameter within the plugin)

Thanks for the tip. Done and documented in the PR!

> Also documentation is very welcome

I updated [Auto-Standby](https://github.com/koreader/koreader/wiki/Auto-Standby) accordingly with:
- A detailed [Purpose of this plugin](https://github.com/koreader/koreader/wiki/Auto-Standby#purpose-of-this-plugin) section
- A section dedicated to the [Allow auto-standby](https://github.com/koreader/koreader/wiki/Auto-Standby#allow-auto-standby) setting
- A [Plugin usage examples](https://github.com/koreader/koreader/wiki/Auto-Standby#plugin-usage-examples) section
  - Though currently it only contains an example I'm familiar with, [Dim the headlight after a period of inactivity](https://github.com/koreader/koreader/wiki/Auto-Standby#dim-the-headlight-after-a-period-of-inactivity)

**Comment by seb-maillard** [comment]
> I wasn't happy with the stock InkPad Auto screen lock settings (Off, 5 and 10 minutes)

By the way, I've submitted a feature request to the PocketBook support to provide more Auto screen lock settings, similar to what the Kindle offers, e.g. 5, 10, 15, 30 and 60 minutes. My request was forwarded to the R&D department. I'll provide an update here depending on how it goes.
